app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/RecipientLayoutCreator.kt

```diff
@@ -83,6 +96,23 @@ internal class RecipientLayoutCreator(
 
         return totalWidth <= availableWidth
     }
+
+    private fun CharSequence.toDisplayRecipients(): CharSequence {
+        return buildSpannedString {
+            append(recipientsPrefix)
+            append(this@toDisplayRecipients)
+            append(recipientsSuffix)
+        }
+    }
+
+    private fun Sequence<CharSequence>.toDisplayRecipients(): CharSequence {
+        return joinTo(
+            buffer = SpannableStringBuilder(),
+            separator = LIST_SEPARATOR,
+            prefix = recipientsPrefix,
+            postfix = recipientsSuffix,
+        ).toSpanned()
+    }
```

Comment:
I guess those two methods could be extracted. Technically, they're not formatting the names (`MessageViewRecipientFormatter` is doing that), they're only creating a list of already formatted strings. So maybe `RecipientListFormatter`?

> Also that one is a recursive function is not ideal.

It's not. I'm guessing the `this@toDisplayRecipients` tripped you up. The qualifier is necessary to refer to the `CharSequence` instance that `toDisplayRecipients()` was called on, because inside the `buildSpannedString { }` block `this` refers to `SpannableStringBuilder`.

Reply:
I cut down the code to just one method, but opted to keep it inside `RecipientLayoutCreator`.